backend/src/main/java/nl/jed/supersimplesupplysystem/security/oauth2/user/OAuth2UserInfoFactory.java

```diff
@@ -0,0 +1,25 @@
+package nl.jed.supersimplesupplysystem.security.oauth2.user;
+
+import nl.jed.supersimplesupplysystem.dto.SocialProvider;
+import nl.jed.supersimplesupplysystem.exception.OAuth2AuthenticationProcessingException;
+
+import java.util.Map;
+
+
+public class OAuth2UserInfoFactory {
```

Comment:
heel veel if else statements, even kijken of we dat om kunnen schrijven naar iets war fijner is om mee te werken